src/traverse.ts

```diff
@@ -27,7 +28,7 @@ export interface Visitor {
   StringLiteral?: Visit<S.StringLiteral>;
   Template?: Visit<S.Template>;
   Ternary?: Visit<S.Ternary>;
-  [propName: string]: Visit<S.Node> | undefined;
+  [propName: string]: Visit<any> | undefined;
```

Comment:
If at all possible, I'd like to avoid having to make this change. I'm not sure what compiler error you are running into, but maybe we can find a different way to fix it? If anything, I think I'd rather find a way to remove this index signature.

Replacing this with `any` is means we lose type safety here. I don't remember exactly why I added it, but I think this index signature was more of a fallback. All of the other `Visitor` attributes implement `Node`, so this unified with those, at least at a more general level.

TLDR: Let's try to keep this as `Visit<S.Node>`, but if we absolutely have no other options, then maybe we can look at other options (like replacing it with `unknown`, `any`, etc.).

Reply:
hey @mthadley, I was also thinking of that, as everyone else also extends from `Node`, would have no problems. But when it extends seems to lose its reference...

But there is an error when trying to compile:
```typescript
[ts] Property 'BooleanLiteral' of type 'VisitFunction<BooleanLiteral> | VisitObject<BooleanLiteral> | undefined' is not assignable to string index type 'VisitFunction<Node> | VisitObject<Node> | undefined'.
(property) Visitor.BooleanLiteral?: VisitFunction<S.BooleanLiteral> | VisitObject<S.BooleanLiteral> | undefined
```

This happens in all `Visitor` properties. We could solve it this way:
```typescript
export type Visit<T> = VisitFunction<keyof T> | VisitObject<keyof T>;
```

But we will be just taking the properties of each interface... I'm thinking of other ways how we can handle this. Do you have any suggestions for not going with `any`?